core/desugarer.cpp

```diff
@@ -63,6 +63,7 @@ BuiltinDecl jsonnet_builtin_decl(unsigned long builtin)
         case 24: return {U"primitiveEquals", {U"a", U"b"}};
         case 25: return {U"native", {U"name"}};
         case 26: return {U"md5", {U"str"}};
+        case 27: return {U"trace", {U"str", U"obj"}};
```

Comment:
Once the trace call can accept any type as the second parameter, what to call this here ? Right now it is obj. 
Does it actually matter except what is printed in an error message ?

Reply:
It doesn't really matter.  `rest` might work.

Another option here is to return `true`, in which case this can be used as in if `std.trace("foo") then ...` or just `assert std.trace("foo"); ...`

Reply:
I do not understand what you suggest. 
First do you want to use "rest" like 
`case 27: return {U"trace", {U"str", U"rest"}};` 
?
Secondly, how would the return true look like ?

Reply:
These are just giving the parameter names on the Jsonnet side, which matter because of being able to do std.md5(str="foo"), and otherwise aren't specified anywhere.  There is no equivalent for the return value, I just mentioned it because I was thinking of ways to avoid the 2nd param, which might get a bit annoying because of the need to put a ) some distance away, when you insert the std.trace.  However my suggestions are also a bit obtuse, since this really has nothing to do with assert or conditionals :)

Reply:
`case 27: return {U"trace", {U"str", U"rest"}};` would be fine

Reply:
👍 for `case 27: return {U"trace", {U"str", U"rest"}};`. Functions that provide alternative semantics can be later implemented as regular jsonnet functions (non-builtin).